app/components/base_component.rb

```diff
@@ -1,17 +1,35 @@
 class BaseComponent < ViewComponent::Base
   def before_render
-    return if @rendered_scripts
-    @rendered_scripts = true
-    if helpers.respond_to?(:enqueue_component_scripts) && self.class.scripts.present?
-      helpers.enqueue_component_scripts(*self.class.scripts)
-    end
+    render_assets unless @has_rendered_assets
   end
 
   def self.scripts
-    @scripts ||= _sidecar_files(['js', 'ts']).map { |file| File.basename(file, '.*') }
+    @scripts ||= sidecar_files_basenames(['js', 'ts'])
+  end
+
+  def self.stylesheets
+    @stylesheets ||= sidecar_files_basenames(['scss'])
   end
 
   def unique_id
     @unique_id ||= SecureRandom.hex(4)
   end
+
+  private
+
+  def self.sidecar_files_basenames(extensions)
+    _sidecar_files(extensions).map { |file| File.basename(file, '.*') }
+  end
+
+  def render_assets
+    if helpers.respond_to?(:enqueue_component_scripts) && self.class.scripts.present?
+      helpers.enqueue_component_scripts(*self.class.scripts)
+    end
+
+    if helpers.respond_to?(:enqueue_component_stylesheets) && self.class.stylesheets.present?
+      helpers.enqueue_component_stylesheets(*self.class.stylesheets)
+    end
+
+    @has_rendered_assets = true
```

Comment:
Feedback welcome on naming, since the prefixing of `has_` would normally be omitted in a method name in favor of the `?` suffix.

Are any of the following (or other ideas) preferable?

```suggestion
    @rendered_assets = true
```

```suggestion
    @rendered_assets = true
  end
    
  def rendered_assets?
    @rendered_assets
```

Reply:
I think it's fine (and clearer) to leave the `has_` in the ivar name, if we needed a method, I would drop it and make the method like:

```ruby
def rendered_assets?
  @has_rendered_assets
end
```

Reply:
> if we needed a method, I would drop it and make the method like:

I like that 👍  Updated in db59bff2c.